pkg/operator/status/status_controller.go

```diff
@@ -83,10 +83,16 @@ func NewClusterOperatorStatusController(
 func (c StatusSyncer) sync() error {
 	detailedSpec, currentDetailedStatus, _, err := c.operatorClient.GetOperatorState()
 	if apierrors.IsNotFound(err) {
-		c.eventRecorder.Warningf("StatusNotFound", "Unable to determine current operator status for %s", c.clusterOperatorName)
-		if err := c.clusterOperatorClient.ClusterOperators().Delete(c.clusterOperatorName, nil); err != nil && !apierrors.IsNotFound(err) {
```

Comment:
While I am not sure this delete makes sense in general, I believe we just need to wait for informer.HasSynced before attempting to use it.

Reply:
I did not know of this HasSynced, looking.

Reply:
I've updated to add the HasSynced, testing now.

Reply:
If operator state is not found, does this mean the ClusterOperator object will stay with invalid contents?

Reply:
I'll look closer, test with this.  It should be fine to keep the delete ifNotFound, as long as we waitForCacheSync.  What I see now is the statusController is up, checks for serviceca CO, doesn't find it bc the informer is not caught up,  that causes a glitch for a few seconds in version reporting (while it recreates?).  The glitch is most likely fixed just by adding the waitForCacheSync though.  I'll report back.

Reply:
is there a reason one would _not_ want the waitForCacheSync? I could be missing something.  : )

Reply:
@sttts @soltysh I confirmed with an upgrade that the WaitForCacheSync resolves the issue I'm seeing with service-ca, authentication COs (last I checked they are only operators other than kube-apiserer making use of this ClusterOperatorStatusController).  I've restored the Delete ifNotFound, I'm still not sure it's necessary but it's not causing the glitch, it's the COInformer not being synced that is.  : )